### ISSUE TITLE
TypeError: nil is not a symbol nor a string

### ISSUE THREAD

**Issue by bgeesaman** [RedisGraph/redisgraph-rb#9]
Hi there.  Was trying to run through the samples and ran into an odd error.  It appears that results without named headers are not happy.  Here's my environment (although I'm pretty sure it has to do with the 2.0.0-2.0.1 changes).  Would love another set of eyes to help me figure this out.  Thanks!

```
$ ruby -v
ruby 2.6.6p146 (2020-03-31 revision 67876) [x86_64-darwin19]
```

```
$ gem list | grep redis
redis (4.2.1)
redisgraph (2.0.1)
```

docker-compose.yml (run via docker-compose up)
```
version: '3'
services:
  db:
    image: docker.io/redislabs/redisgraph:2.0.20 # doesn't work with 2.0.0-2.0.20
    hostname: db
    container_name: db
    ports:
      - "6379:6379"
```

Running `rspec` from inside the gem directory:
```
$ rspec
fatal: not a git repository (or any of the parent directories): .git
......F.F

Failures:

  1) RedisGraph edges should traverse edges properly
     Failure/Error: val.send(map_func)
     
     TypeError:
       nil is not a symbol nor a string
     # ./lib/redisgraph/query_result.rb:98:in `map_scalar'
     # ./lib/redisgraph/query_result.rb:68:in `block (2 levels) in parse_resultset'
     # ./lib/redisgraph/query_result.rb:65:in `each'
     # ./lib/redisgraph/query_result.rb:65:in `reduce'
     # ./lib/redisgraph/query_result.rb:65:in `block in parse_resultset'
     # ./lib/redisgraph/query_result.rb:63:in `map'
     # ./lib/redisgraph/query_result.rb:63:in `parse_resultset'
     # ./lib/redisgraph/query_result.rb:36:in `initialize'
     # ./lib/redisgraph.rb:61:in `new'
     # ./lib/redisgraph.rb:61:in `query'
     # ./spec/redisgraph_spec.rb:74:in `block (3 levels) in <top (required)>'

  2) RedisGraph update should print property strings correctly after updates
     Failure/Error: val.send(map_func)
     
     TypeError:
       nil is not a symbol nor a string
     # ./lib/redisgraph/query_result.rb:98:in `map_scalar'
     # ./lib/redisgraph/query_result.rb:68:in `block (2 levels) in parse_resultset'
     # ./lib/redisgraph/query_result.rb:65:in `each'
     # ./lib/redisgraph/query_result.rb:65:in `reduce'
     # ./lib/redisgraph/query_result.rb:65:in `block in parse_resultset'
     # ./lib/redisgraph/query_result.rb:63:in `map'
     # ./lib/redisgraph/query_result.rb:63:in `parse_resultset'
     # ./lib/redisgraph/query_result.rb:36:in `initialize'
     # ./lib/redisgraph.rb:61:in `new'
     # ./lib/redisgraph.rb:61:in `query'
     # ./spec/redisgraph_spec.rb:91:in `block (3 levels) in <top (required)>'

Finished in 0.03635 seconds (files took 0.32872 seconds to load)
9 examples, 2 failures

Failed examples:

rspec ./spec/redisgraph_spec.rb:70 # RedisGraph edges should traverse edges properly
rspec ./spec/redisgraph_spec.rb:89 # RedisGraph update should print property strings correctly after updates
```

I discovered this when in my test app I was trying to fetch a single node:
```
pry> dbconn.query('create (p:Person {name: "Andy"})')
pry> dbconn.query('MATCH (p {name: "Andy"}) RETURN p.name').resultset
=> [["Andy"]]
pry> dbconn.query('MATCH (p {name: "Andy"}) RETURN p.name as name').print_resultset
+------+
| name |
+------+
| Andy |
+------+
pry> dbconn.query('MATCH (p {name: "Andy"}) RETURN p').resultset
TypeError: nil is not a symbol nor a string
from /Users/myusername/.rvm/rubies/ruby-2.6.6/lib/ruby/gems/2.6.0/gems/redisgraph-2.0.1/lib/redisgraph/query_result.rb:98:in `map_scalar'
```

**Comment by bgeesaman** [comment]
Yep, gem 2.0.0:
```
pry> dbconn.query('MATCH (p {name: "Andy"}) RETURN p').resultset
=> []
```

**Comment by jeffreylovitz** [comment]
Hey @bgeesaman,

I've confirmed this bug; it looks like I didn't test compatibility with the RedisGraph 2.0.x family properly. The tests pass with the latest version of RedisGraph, which you can use with the `edge` Docker tag.

I wouldn't use 2.0.0, as it has a bug that causes it to not retrieve data like property keys and label names properly.

I can probably write a fix for this next week! Or, if you're interested in contributing, I have a pretty good idea of what the issue is if you want to give it a shot.

**Comment by bgeesaman** [comment]
@jeffreylovitz Ah, switching to the `docker.io/redislabs/redisgraph:edge` seems to work.  Thank you for that guidance and the heads up about not using `2.0.0`.  I believe I pinpointed the issue in the code, but unfortunately, I don't know that I could confidently address it in any reasonable amount of time.  Next week would be fantastic.  You have me unstuck at the moment, so thank you!

**Comment by jeffreylovitz** [comment]
This issue has been resolved in the new gem version 2.0.2.

**Comment by bgeesaman** [comment]
Thank you!
